tests/test_core.py

```diff
@@ -7207,6 +7207,7 @@ def test_async(self, emterpretify=False):
 
     if emterpretify:
       print('check bad ccall use')
+      self.set_setting('EXTRA_EXPORTED_RUNTIME_METHODS', ["ccall"])
```

Comment:
Given that `ccall` and `cwrap` are now regular JS library functions, wouldn't it be appropriate to use the regular JS library function export mechanism (`-s DEFAULT_LIBRARY_FUNCS_TO_INCLUDE`(?)) to export them? Using the special mechanism `EXTRA_EXPORTED_RUNTIME_METHODS` (which iiuc was crafted just for the preamble functions) to export `ccall` after it has been "demoted" from being a special preamble function to a regular JS library function looks like an odd direction?

Reply:
Does the above response address this?

Reply:
I think we should wither `EXTRA_EXPORTED_RUNTIME_METHODS` away as we move functions away from preamble/runtime, since the meaning of `EXTRA_EXPORTED_RUNTIME_METHODS` was crafted specifically to cater to the special symbols found in preamble/runtime. As these symbols convert from special to regular JS library symbols, it would be good to enforce their linkage to be treated like other regular JS library symbols should be.

As backwards compatibility for `ccall` specifically and other existing symbols that were part of runtime/preamble, if we want to keep supporting the usage, we could, but I'd rather we enforced that users didn't.

Perhaps we should build a new machinery to link+export JS library functions, like described in the other comment above? It feels like perhaps we have been missing a construct here. Would that solve the issue cleanly for all JS library items?